core/connection/stub_connection.go

```diff
@@ -39,7 +43,7 @@ type connectionFactoryFake struct {
 	mockConnection *connectionFake
 }
 
-func (cff *connectionFactoryFake) CreateConnection(connectionParams ConnectOptions, stateChannel StateChannel) (Connection, error) {
+func (cff *connectionFactoryFake) CreateConnection(connectionParams ConnectOptions, stateChannel StateChannel, statisticsChannel StatisticsChannel) (Connection, error) {
```

Comment:
I'll move this to the stubs file (each and every single one of those had a separate file in the core/connection), but I'd rather not rewrite tests along the pull request that changes a whole lot in the package it's testing. Since the tests were stable for a long time, minimal changes to them assures that one does not break anything with the PR. It might be a candidate for a future refactor though.